src/bezpath.rs

```diff
@@ -1108,10 +1108,10 @@ impl From<QuadBez> for PathSeg {
 }
 
 impl Shape for BezPath {
-    type PathElementsIter = std::vec::IntoIter<PathEl>;
+    type PathElementsIter<'iter> = std::iter::Copied<std::slice::Iter<'iter, PathEl>>;
 
-    fn path_elements(&self, _tolerance: f64) -> Self::PathElementsIter {
-        self.0.clone().into_iter()
+    fn path_elements(&self, _tolerance: f64) -> Self::PathElementsIter<'_> {
+        self.0.iter().copied()
```

Comment:
so the question I have with all of these is basically: do we even bother with `copy`? We we need to?

Reply:
Do you mean "why do we copy the elements rather than returning a ref to them? If so then the answer is because some impls of `path_elements` will create the path elements on-demand, and so need to pass ownership of them to the callee.

If I've misunderstood let me know.

Reply:
Right, that answers my question. LGTM!

Reply:
I think it's an important question. From a rough calculation, the size of `PathEl` is around 50 bytes - not insignificant! 

Calc: `Point = 2 x f64 = 16 bytes, PathEl = 3 x Point + u8 (discriminant) = 49 ~ 50`.

Reply:
Measurement is important but I strongly suspect that the optimizer will eliminate these copies.